### PR TITLE
#171 shade all the Dependencies.

### PR DESCRIPTION
**Pull request by firemonk9**
#171 shade all the Dependencies.

### CODE REVIEW

**Comment by firemonk9** [comment]
increasing the scope.

File: src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBSpark.scala [hunk 1]
```diff
@@ -393,7 +353,10 @@ object CosmosDBSpark extends LoggingTrait {
                                          config: Config,
                                          partitionCount: Int,
                                          collectionThroughput: Int): Iterator[D] = {
+    //TODO REMOVE THIS LINE
```
Comment (dennyglee) [inline src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBSpark.scala]: Per the comment, should this line be removed?

File: pom.xml [hunk 2]
```diff
@@ -17,418 +17,621 @@ limitations under the License.
...
-    </developer>
-  </developers>
-  <scm>
-    <connection>scm:git:git@github.com:Azure/azure-cosmosdb-spark.git</connection>
-    <developerConnection>scm:git:git@github.com:Azure/azure-cosmosdb-spark.git</developerConnection>
-    <url>git@github.com:Azure/azure-cosmosdb-spark.git</url>
-  </scm>
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.microsoft.azure</groupId>
+    <artifactId>azure-cosmosdb-spark_2.2.0_2.11</artifactId>
```
Comment (dennyglee) [inline pom.xml]: For master, should we focus on Spark 2.2.1  (matching default Databricks Runtime 3.5) or Spark 2.2.0 (HDI 3.7, Databricks Runtime 3.4).  Note, that we should have a Spark 2.3 version  of this in the Spark 2.3 branch  (matching Databricks Runtime 4.0).

File: pom.xml [hunk 3]
```diff
@@ -17,418 +17,621 @@ limitations under the License.
...
+            <groupId>com.microsoft.azure</groupId>
+            <artifactId>azure-documentdb</artifactId>
+            <version>1.16.0</version>
+        </dependency>
+        <dependency>
+            <groupId>io.reactivex</groupId>
+            <artifactId>rxjava</artifactId>
+            <version>1.3.0</version>
+        </dependency>
+        <!--<dependency>-->
```
Comment (dennyglee) [inline pom.xml]: Since this is commented out, should we just remove it?

**Pull request closed without merging**